### ISSUE TITLE
Allow different object names for speech and braille

### ISSUE THREAD

**Issue by leonardder** [nvaccess/nvda#7335]
In the current situation, objects have only one name property which is used for both speech and braille. I see several use cases for a separate object property (e.g. nameBrl) to allow overriding the object name.

1. Excel selection objects: For these objects, the name looks like "A1  through D1". It would be much shorter if we had something like a1/d1. Same applies to merged cells
2. For sysListView32, list columns are separated with a ; (semicolon) and a space. In braille, we could use a different unicode cell separator for this. Also, we probably should leave out the space.
3. There might be cases where an appmodule writer would like to abbreviate the name in braille, or even discard it.

cc @DKager

**Comment by ruifontes** [comment]
By the way, I don't know if it is related...
The information in a message list of Unread, Attachment, High Priority, also 
should be make shorter in Braille...

Rui Fontes


-----Mensagem Original----- 
De: Leonard de Ruijter
Data: 28 de junho de 2017 16:33
Para: nvaccess/nvda
Cc: Subscribed
Assunto: [nvaccess/nvda] Allow different object names for speech and braille 
(#7335)



In the current situation, objects have only one name property which is used 
for both speech and braille. I see several use cases for a separate object 
property (e.g. nameBrl) to allow overriding the object name.

Excel selection objects: For these objects, the name looks like "A1 through 
D1". It would be much shorter if we had something like a1/d1. Same applies 
to merged cells
For sysListView32, list columns are separated with a ; (semicolon) and a 
space. In braille, we could use a different unicode cell separator for this. 
Also, we probably should leave out the space.
There might be cases where an appmodule writer would like to abbreviate the 
name in braille, or even discard it.

cc @dkager

**Comment by leonardder** [comment]
> The information in a message list of Unread, Attachment, High Priority, also should be make shorter in Braille...

Hmm, this is somehow related, but probably quite difficult to do, as the information to shorten is language specific. In Thunderbird, for english, "Has Attachment" is printed, whereas in Dutch, we have "Heeft bijlage".

**Comment by bramd** [comment]
@leonardder Yes, this is hard. In Thunderbird the accessible name of the whole row is a concatination of all the fields. You could present each field differently in an appmodule, like we do in Outlook, but you would need to find a language independent identifier.

A more generic solution I can imagine and which could be distributed in an add-on is an inteface to customize the reading of listviews. E.g. define a speech and braille dictionary per column (unread -> unrd) etc and select which columns you want to see in which order. If I remember correctly, JAWS has a similar feature to customize lists. That being said, I think that is another issue and this issue of object names is valid on it's own and would be a dependency for the functionality I described.

**Comment by derekriemer** [comment]
[**@ruifontes**](https://github.com/ruifontes) commented on [Jun 28, 2017, 9:48 AM MDT](https://github.com/nvaccess/nvda/issues/7335#issuecomment-311702205 "2017-06-28T15:48:11Z - Replied by Github Reply Comments"):
> By the way, I don't know if it is related...
> The information in a message list of Unread, Attachment, High Priority, also 
> should be make shorter in Braille...

Do we actually have a "attachment" state?

**Comment by ruifontes** [comment]
Hello!

Derek wrote:
Do we actually have a "attachment" state?

Maybe I don't make myself very clear...
In a message list, each message can have the attributes of:
Read/Unread
Attachment
Priority
...

Writing all those words in a Braille display we don't have enough cells to 
the message itself...

Unread, Attachment, High priority,

We already have 33 cells occupied... 7 to display Name, Subject, Date...

Regards,

Rui

**Comment by derekriemer** [comment]
Sure. I was curious if there's an attachment state in accessibility APIs, because that would make this much easier.

**Comment by ruifontes** [comment]
That I don't know!
And it should depends on mail manager app used...

Rui


-----Mensagem Original----- 
De: Derek Riemer
Data: 11 de julho de 2017 12:52
Para: nvaccess/nvda
Cc: ruifontes ; Comment
Assunto: Re: [nvaccess/nvda] Allow different object names for speech and 
braille (#7335)



Sure. I was curious if there's an attachment state in accessibility APIs, 
because that would make this much easier.
